### PR TITLE
Update `Avatar`

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Resolves https://github.com/Shopify/polaris-summer-editions/issues/102

### WHAT is this pull request doing?

- Updates Avatar shape to only be square
- Updates Avatar colors
- Updates Avatar border-radius
- Updates Avatar XS size
- ~Updates Avatar font-sizes~*
- Adds experimental color tokens for decorative colors
- Updates stories to better consolidate examples

_*Right now font-size in the Avatars is scaled based on the medium avatar font-size. Updating this size to the new mocks would overflow the backgrounds currently. And it would be too involved to reconfigure the component to have non-scaled font-sizes for the current scope of work. If it ends up looking off in other components we can definitely tweak this value. When we rework/revisit the Avatar in the future we can explore having non-scaled font-sizes._ 

### How to 🎩

Review on [Storybook](https://5d559397bae39100201eedc1-pwakwmvjdb.chromatic.com/?path=/story/all-components-avatar--all&globals=polarisSummerEditions2023:true)
- Ensure beta flagged Avatar styles render as expected
- Ensure no regressions to the existing Avatar styles

### CODE REVIEW

**Review by kyledurand — APPROVED** [review]
👏

File: polaris-react/src/components/Avatar/Avatar.scss [hunk 1]
```diff
@@ -1,9 +1,12 @@
+@import '../../styles/common';
+
 .Avatar {
   // stylelint-disable -- Polaris component custom properties
   --pc-avatar-extra-small-size: 24px;
   --pc-avatar-small-size: 32px;
   --pc-avatar-medium-size: 40px;
   --pc-avatar-large-size: 60px;
+
```
Comment (aaronccasanova) [inline polaris-react/src/components/Avatar/Avatar.scss]:
```suggestion
```

File: polaris-react/src/components/Avatar/Avatar.scss [hunk 2]
```diff
@@ -82,34 +104,114 @@
...
+    #{$se23} & {
+      color: var(--p-color-avatar-icon-yellow-experimental);
+    }
+  }
+
+  text {
+    #{$se23} & {
+      color: var(--p-color-avatar-text-yellow-experimental);
+    }
+  }
```
Comment (aaronccasanova) [inline polaris-react/src/components/Avatar/Avatar.scss:121]:
```suggestion
  #{$se23} & svg,
  #{$se23} & text {
    color: var(--p-color-avatar-icon-yellow-experimental);
  }
```

Light suggestion

**Review by aaronccasanova — APPROVED** [review]

File: polaris-react/src/components/Avatar/Avatar.scss (hunk 2 above)
Comment (aaronccasanova) [reply polaris-react/src/components/Avatar/Avatar.scss:121]: Also applicable to `.styleTwo` through `.styleFive`

Comment (lgriffee) [reply polaris-react/src/components/Avatar/Avatar.scss:121]: These are applying two different colors to icons vs. text so I don't think they can be combined. 

- `--p-color-avatar-icon-yellow-experimental`
- `--p-color-avatar-text-yellow-experimental`

Comment (aaronccasanova) [reply polaris-react/src/components/Avatar/Avatar.scss:121]: Ah yes! Missed that subtle nuance 👍

**Review by sarahill — APPROVED** [review]

**Pull request merged**